sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2ScanRelationPushDown.scala

```diff
@@ -87,9 +87,26 @@ object V2ScanRelationPushDown extends Rule[LogicalPlan] with PredicateHelper {
           if filters.isEmpty && project.forall(_.isInstanceOf[AttributeReference]) =>
           sHolder.builder match {
             case r: SupportsPushDownAggregates =>
+              val normalizedGroupingExpressions = DataSourceStrategy.normalizeExprs(
+                groupingExpressions, sHolder.relation.output)
+              val translatedGroupBys =
+                PushDownUtils.translateAggregation(Seq.empty, normalizedGroupingExpressions)
```

Comment:
why can't we put the real aggregate functions here?

Reply:
Because the aggregates are collected from Aggregate. I want call r.supportCompletePushDown and split AVG into 2 functions: SUM and COUNT first, so we collect AggregateExpression just once.

Reply:
This assumes the v2 source only need the group by columns to decide if it supports complete pushdown or not. Do you have strong evidence to indicate it's true for all v2 sources?

Reply:
Let's pass aggregate expressions too for all the DS V2.